web_src/js/features/comp/EditorMarkdown.js

```diff
@@ -0,0 +1,100 @@
+import {triggerEditorContentChanged} from './Paste.js';
+
+function handleIndentSelection(textarea, e) {
+  const selStart = textarea.selectionStart;
+  const selEnd = textarea.selectionEnd;
+  if (selEnd === selStart) return; // do not process when no selection
+
+  e.preventDefault();
+  const lines = textarea.value.split('\n');
+  const selectedLines = [];
+
+  let pos = 0;
+  for (let i = 0; i < lines.length; i++) {
+    if (pos > selEnd) break;
+    if (pos >= selStart) selectedLines.push(i);
+    pos += lines[i].length + 1;
+  }
+
+  for (const i of selectedLines) {
+    if (e.shiftKey) {
+      lines[i] = lines[i].replace(/^(\t| {1,2})/, '');
+    } else {
+      lines[i] = `  ${lines[i]}`;
+    }
+  }
+
+  // re-calculating the selection range
+  let newSelStart, newSelEnd;
+  pos = 0;
+  for (let i = 0; i < lines.length; i++) {
+    if (i === selectedLines[0]) {
+      newSelStart = pos;
+    }
+    if (i === selectedLines[selectedLines.length - 1]) {
+      newSelEnd = pos + lines[i].length;
+      break;
+    }
+    pos += lines[i].length + 1;
+  }
+  textarea.value = lines.join('\n');
+  textarea.setSelectionRange(newSelStart, newSelEnd);
+  triggerEditorContentChanged(textarea);
+}
+
+function handleNewline(textarea, e) {
+  const selStart = textarea.selectionStart;
+  const selEnd = textarea.selectionEnd;
+  if (selEnd !== selStart) return; // do not process when there is a selection
+
+  const value = textarea.value;
+
+  // find the current line
+  const lineStart = value.lastIndexOf('\n', selStart - 1) + 1;
```

Comment:
I can see two edge cases here:
- `\n` not present -> only works because it will probably return `-1` which is the intended `0` when adding one
- when you select exactly until the beginning of a line, I think we have an off-by-one error as we consider the wrong line or two lines at the same time?

Reply:
please show a real case.

Reply:
Let's assume the following text:
```txt
a line\n
0123456
next
78910
```
I'm selecting now from `7` (selStart) to `8` (selEnd)
Thus, `lineStart == 6 + 1`

… and no I do see that it works as expected.